src/pkg/services/m365/api/teams.go

```diff
@@ -0,0 +1,165 @@
+package api
+
+import (
+	"context"
+
+	"github.com/alcionai/clues"
+	msgraphgocore "github.com/microsoftgraph/msgraph-sdk-go-core"
+	"github.com/microsoftgraph/msgraph-sdk-go/models"
+
+	"github.com/alcionai/corso/src/internal/common/str"
+	"github.com/alcionai/corso/src/internal/common/tform"
+	"github.com/alcionai/corso/src/internal/m365/graph"
+	"github.com/alcionai/corso/src/pkg/fault"
+	"github.com/alcionai/corso/src/pkg/logger"
+)
+
+const (
+	teamsAdditionalDataLabel    = "Team"
+	ResourceProvisioningOptions = "resourceProvisioningOptions"
+)
+
+// ---------------------------------------------------------------------------
+// controller
+// ---------------------------------------------------------------------------
+
+func (c Client) Teams() Teams {
+	return Teams{c}
+}
+
+// On creation of each Teams team a corrsponding group gets created.
+// The group acts as the protected resource, and all teams data like events,
+// drive and mail messages are owned by that group.
+
+// Teams is an interface-compliant provider of the client.
+type Teams struct {
+	Client
+}
+
+// GetAllTeams retrieves all groups.
+func (c Teams) GetAll(
+	ctx context.Context,
+	errs *fault.Bus,
+) ([]models.Groupable, error) {
+	service, err := c.Service()
+	if err != nil {
+		return nil, err
+	}
+
+	return getGroups(ctx, true, errs, service)
+}
+
+// GetAll retrieves all groups.
+func getGroups(
+	ctx context.Context,
+	getOnlyTeams bool,
```

Comment:
As mentioned in the other comment, let's ditch this for now and get all groups instead.

That said, not all groups have access to conversations (and, I assume, groups without access to conversations also don't have access to a calendar).  Instead of packing that filter into the pagination, how about a separate func that can accept a `[]models.Groupable` as a parameter and filter it down to only specific groups, such as only those which can access conversations?  This change can be made in a follow-up PR.

Reply:
Done. Currently we have only one such func implemented for teams. But it can be extended in future.